### PR TITLE
Profile page

### LINKED ISSUE #330: Create profile page
- [ ] Convert students page and teachers page to profile page. 
- [ ] Add button "Profile" to right-top menu. After clicking on it user will get to his own student or teacher page. If teacher wants to open any students or teachers page he'll get to his profile page. 
- [ ] Add changing password button to the profile page

### PR DESCRIPTION
**Pull request by askatsevalov**
Closes #330

### CODE REVIEW

File: back-end/src/auth/serializers.py [hunk 1]
```diff
@@ -93,12 +95,37 @@ class UserSerializer(serializers.ModelSerializer):
...
+        # check if user is a teacher or a student
+        user = Teacher.objects.filter(user=user_to_check)
+        user_type = "teacher"
+        if user.count() == 0:
+            # check if user is a student
+            user = Student.objects.filter(user=user_to_check)
+            user_type = "student"
+            if user.count() == 0:
+                return "", 0
+        return user_type, user.first().id
```
Comment (TmLev) [inline back-end/src/auth/serializers.py:118]: @gakhromov I've seen this code already.

File: front-end/src/components/Personnel/Teacher/Teacher.vue [hunk 2]
```diff
@@ -205,71 +306,144 @@ export default {
...
+      };
+      reader.readAsDataURL(file);
+
+      return true;
     },
     backToPersonnel() {
-      this.$router.push({ path: "/personnel/" });
+      this.$router.push({ name: "Personnel" });
+    },
+    beforeClose() {
```
Comment (i-oktav-i) [inline front-end/src/components/Personnel/Teacher/Teacher.vue:433]: async

File: front-end/src/components/Personnel/Student/StudentGeneral/StudentGeneral.vue [hunk 3]
```diff
@@ -203,21 +206,75 @@ export default {
...
+                callback();
+              }
+            },
+            trigger: "change",
+          },
+        ],
+        birth_info: [
+          {
+            validator: (rule, value, callback) => {
+              console.log("🚀 > value", value);
```
Comment (TmLev) [inline front-end/src/components/Personnel/Student/StudentGeneral/StudentGeneral.vue:232]: Remove.

File: front-end/src/components/Marks/Marks.vue [hunk 4]
```diff
@@ -132,31 +136,28 @@
...
                             effect="dark"
                             disable-transitions
                             class="is-clickable margin-x"
-                            @click="onEdit(
-                              data.marks.find((x) => x.lesson === item.id),
-                              data,
-                            )"
+                            @click="
+                              onEdit(
+                                data.marks.find((x) => x.lesson === item.id),
```
[the remark this answers is not in the document]
Comment (askatsevalov) [reply front-end/src/components/Marks/Marks.vue:157]: нет

**Review by TmLev — APPROVED** [review]
LGTM.

**Pull request merged**